addons/l10n_it_edi_doi/views/l10n_it_edi_doi_declaration_of_intent_views.xml

```diff
@@ -0,0 +1,140 @@
+                <field name="company_id" groups="base.group_multi_company" optional="hidden"/>
+                <field name="protocol_number_part1" readonly="state != 'draft'"/>
+                <field name="protocol_number_part2" readonly="state != 'draft'"/>
+                <field name="issue_date" readonly="state != 'draft'"/>
+                <field name="start_date" readonly="state != 'draft'"/>
+                <field name="end_date" readonly="state != 'draft'"/>
+                <field name="threshold" readonly="state != 'draft'"/>
+                <field name="currency_id" readonly="state != 'draft'" optional="hidden"/>
+                <field name="not_yet_invoiced" optional="hidden"/>
+                <field name="invoiced" optional="hidden"/>
+                <field name="remaining"/>
+                <field name="state"/>
+            </tree>
+        </field>
+    </record>
+
+    <record id="view_l10n_it_edi_doi_declaration_of_intent_form" model="ir.ui.view">
+        <field name="name">l10n_it_edi_doi.declaration_of_intent.form</field>
+        <field name="model">l10n_it_edi_doi.declaration_of_intent</field>
+        <field name="arch" type="xml">
+            <form>
+                <header>
+                    <field name="state" widget="statusbar" statusbar_visible="draft,active"/>
+                    <button string="Validate" invisible="state != 'draft'" class="btn-primary"
+                            type="object" name="action_validate"/>
+                    <button string="Terminate" invisible="state != 'active'" class="btn-primary"
+                            type="object" name="action_terminate"/>
+                    <button string="Reset to Draft" invisible="state != 'active'" class="btn-secondary"
+                            type="object" name="action_reset_to_draft"/>
+                    <button string="Reactivate" invisible="state not in ['terminated', 'revoked']" class="btn-secondary"
+                            type="object" name="action_reactivate"/>
+                    <button string="Revoke" invisible="state != 'active'" class="btn-secondary"
+                            type="object" name="action_revoke"/>
+                </header>
+                <sheet>
+                    <div class="oe_button_box" name="button_box">
+                        <field name="invoice_ids" invisible="True"/>
+                        <button type="object"
+                                class="oe_stat_button"
+                                name="action_open_invoice_ids"
+                                icon="fa-pencil-square-o"
+                                invisible="not invoice_ids">
+                            <div class="o_field_widget o_stat_info">
+                                <span class="o_stat_text">Invoices</span>
+                            </div>
+                        </button>
+                        <field name="sale_order_ids" invisible="True"/>
```

Comment:
It's usually a good practice to group the invisible fields together at the top :wink:

Reply:
I put it at the start of the <div> or did you rather mean at the start of the <sheet>?